extensions/assets-manager/AssetsManagerEx.cpp

```diff
@@ -476,11 +498,14 @@ void AssetsManagerEx::parseVersion()
         }
         else
         {
+            // Save the old update entry type since its value may be changed in event listener.
+            UpdateEntry oldUpdateEntry = _updateEntry;
+
             _updateState = State::NEED_UPDATE;
             dispatchUpdateEvent(EventAssetsManagerEx::EventCode::NEW_VERSION_FOUND);
 
             // Wait to update so continue the process
-            if (_waitToUpdate)
+            if (oldUpdateEntry == UpdateEntry::DO_UPDATE)
```

Comment:
Yes, I understand, I mean if it's in checkUpdate, then _updateEntry will be set to NONE, then it's ok to verify _updateEntry here, no need to use oldUpdateEntry